spring-cloud-config-client/src/test/java/org/springframework/cloud/config/client/ConfigClientPropertiesTests.java

```diff
@@ -59,7 +59,42 @@ public void explicitPassword() {
 		assertEquals("foo", locator.getUsername());
 		assertEquals("secret", locator.getPassword());
 	}
+	
+	@Test
+	public void testIfNoColonPresentInUriCreds() {
+		locator.setUri("http://foobar@localhost:9999");
+		locator.setPassword("secret");
+		assertEquals("http://localhost:9999", locator.getRawUri());
+		assertEquals("foobar", locator.getUsername());
+		assertEquals("secret", locator.getPassword());
+	}
 
+	@Test
+	public void testIfColonPresentAtTheEndInUriCreds() {
+		locator.setUri("http://foobar:@localhost:9999");
+		locator.setPassword("secret");
+		assertEquals("http://localhost:9999", locator.getRawUri());
+		assertEquals("foobar", locator.getUsername());
+		assertEquals("secret", locator.getPassword());
+	}
+	
+	@Test
+	public void testIfColonPresentAtTheStartInUriCreds() {
+		locator.setUri("http://:foobar@localhost:9999");
+		assertEquals("http://localhost:9999", locator.getRawUri());
+		assertEquals("user", locator.getUsername());
```

Comment:
This test is invalid the user in this case should be the empty string since no user was specified.  The first colon splits the username from the password.

Reply:
How about if no user was specified, keeping the default value of username which is user?